tla-io/src/test/scala/at/forsyte/apalache/io/quint/TestQuintEx.scala

```diff
@@ -439,4 +452,53 @@ class TestQuintEx extends AnyFunSuite {
   test("can convert builtin assert operator") {
     assert(convert(Q.app("assert", Q.nIsGreaterThanZero)) == "n > 0")
   }
+
+  test("can convert builtin Map operator") {
+    assert(convert(Q.app("Map", Q.intTup1, Q.intTup2)) == "Apalache!SetAsFun({<<0, 1>>, <<3, 42>>})")
+  }
+
+  test("can convert builtin get operator") {
+    assert(convert(Q.app("get", Q.intMap, Q._0)) == "Apalache!SetAsFun({<<0, 1>>, <<3, 42>>})[0]")
+  }
+
+  test("can convert builtin keys operator") {
+    assert(convert(Q.app("keys", Q.intMap)) == "DOMAIN Apalache!SetAsFun({<<0, 1>>, <<3, 42>>})")
+  }
+
+  test("can convert builtin setToMap operator") {
+    assert(convert(Q.app("setToMap", Q.intPairSet)) == "Apalache!SetAsFun({<<1, 2>>, <<1, 2>>})")
+  }
+
+  test("can convert builtin setOfMaps operator") {
+    assert(convert(Q.app("setOfMaps", Q.intSet, Q.intSet)) == "[{1, 2, 3} → {1, 2, 3}]")
+  }
+
+  test("can convert builtin set operator") {
+    assert(
+        convert(Q.app("set", Q.intMap, Q._1, Q._42))
+          ==
+            "[Apalache!SetAsFun({<<0, 1>>, <<3, 42>>}) EXCEPT ![1] = 42]"
+    )
+  }
+
+  test("can convert builtin mapBy operator") {
+    assert(convert(Q.app("mapBy", Q.intSet, Q.addOneOp)) == "[n ∈ {1, 2, 3} ↦ n + 1]")
+  }
+
+  test("can convert builtin setBy operator") {
+    val expected = """
+        |[Apalache!SetAsFun({<<0, 1>>, <<3, 42>>}) EXCEPT ![1] =
+        |(LET __QUINT_LAMBDA0(n) ≜ n + 1 IN __QUINT_LAMBDA0(Apalache!SetAsFun({<<0, 1>>, <<3, 42>>})[1]))]
```

Comment:
```suggestion
        |[Apalache!SetAsFun({<<0, 1>>, <<3, 42>>}) EXCEPT ![3] =
        |(LET __QUINT_LAMBDA0(n) ≜ n + 1 IN __QUINT_LAMBDA0(Apalache!SetAsFun({<<0, 1>>, <<3, 42>>})[3]))]
```

Reply:
Not entirely sure how the caching of LET-INs works, but would it make sense to cache the `SetAsFun(_)` occuring twice here? /cc @konnov

Reply:
Ah good point. I should have considered that. I'll cache it.